src/Encoding/JoseEncoder.php

```diff
@@ -48,18 +47,15 @@ public function jsonDecode(string $json)
 
     public function base64UrlEncode(string $data): string
     {
-        return rtrim(strtr(base64_encode($data), '+/', '-_'), '=');
+        return sodium_bin2base64($data, SODIUM_BASE64_VARIANT_URLSAFE_NO_PADDING);
     }
 
     public function base64UrlDecode(string $data): string
     {
-        // Padding isn't added back because it isn't strictly necessary for decoding with PHP
-        $decodedContent = base64_decode(strtr($data, '-_', '+/'), true);
-
-        if (! is_string($decodedContent)) {
+        try {
+            return sodium_base642bin($data, SODIUM_BASE64_VARIANT_URLSAFE_NO_PADDING, '');
+        } catch (SodiumException $sodiumException) {
             throw CannotDecodeContent::invalidBase64String();
```

Comment:
We should forward the previous exception then, that would be of great help for folks.

Reply:
I thought about it: is it ok for you to accept for `CannotDecodeContent::invalidBase64String` an optional exception argument, so we can keep the API while forwarding the sodium exception?

Reply:
Sure thing

Reply:
Turned out `CannotDecodeContent::invalidBase64String` is only used with sodium now, so I've enforced the API with the sodium exception

Reply:
Yeap, it's a minor BC-break but it's quite acceptable. Folks will only catch the exception.